test/views/RepresentativeDanViewSpec.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import base.ViewBaseSpec
+import forms.RepresentativeDanFormProvider
+import messages.{BaseMessages, RepresentativeDanMessages}
+import models.RepresentativeDan
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.data.Form
+import play.api.mvc.Call
+import play.twirl.api.Html
+import views.html.RepresentativeDanView
+
+class RepresentativeDanViewSpec extends ViewBaseSpec with BaseMessages {
+
+  def formDataSetup(accountNumber: Option[String] = Some("1234567"),
+                    danType: Option[String] = Some("A")): Map[String, String] =
+    (
+      accountNumber.map(_ => "accountNumber" -> accountNumber.get) ++
+        danType.map(_ => "value" -> danType.get)
+      ).toMap
+
+  private lazy val injectedView: RepresentativeDanView = app.injector.instanceOf[RepresentativeDanView]
+
+  val formProvider: RepresentativeDanFormProvider = injector.instanceOf[RepresentativeDanFormProvider]
+
+  val backLinkUrl = "backLinkUrl"
+
+  "Rendering the Representative Dan One view" when {
```

Comment:
While this is a very clever bit of coding which I must applaud you on, it is also a really complicated way of exercising the error conditions that makes the test difficult to read. I think it would be better specifically having a 'should' clause for each error scenario so it is easier for the next person to understand.
Additionally you have no test for the scenario where there isn't an error.